cache/blobs.go

```diff
@@ -106,13 +108,37 @@ func computeBlobChain(ctx context.Context, sr *immutableRef, createIfNeeded bool
 			if release != nil {
 				defer release()
 			}
-			desc, err := sr.cm.Differ.Compare(ctx, lower, upper,
-				diff.WithMediaType(mediaType),
-				diff.WithReference(sr.ID()),
-				diff.WithCompressor(compressorFunc),
-			)
-			if err != nil {
-				return nil, err
+			var desc ocispecs.Descriptor
+			if len(lower) > 0 && !isTypeWindows(sr) {
+				// Try optimized diff for overlayfs
+				computed, ok, err := sr.tryComputeOverlayBlob(ctx, lower, upper, mediaType, sr.ID(), compressorFunc)
+				if err != nil {
+					logrus.Warnf("failed to compute blob (%s) from diff of overlay snapshotter: %+v", sr.ID(), err)
+				}
+				if ok {
+					desc = computed
+				}
+				if desc.Digest == "" {
+					switch sr.cm.ManagerOpt.Snapshotter.Name() {
+					case "overlayfs", "fuse-overlayfs", "stargz":
+						if os.Getenv("BUILDKIT_DEBUG_FORCE_OVERLAY_DIFF") == "1" {
```

Comment:
this condition should be before `switch`. If overlay diff is forced but native snapshotter is used then it is an error as well.

Reply:
it is better if these env use `ParseBool` instead of `1|0` constants. use a helper method to avoid duplication if reused.

Reply:
Thank you for the suggestion. Implemented the logic.